github3/repos/branch.py

```diff
@@ -30,3 +56,32 @@ def _update_attributes(self, branch):
 
     def _repr(self):
         return '<Repository Branch [{0}]>'.format(self.name)
+
+    def configure(self, enable, enforcement=None, status_checks=None):
+        """Set branch protection and status check enforcement.
+
+        See: http://git.io/v4Gvu
+
+        :param bool enable: (required), True enables protection for this
+            branch. False, disables. Set to None to leave unchanged.
+        :param str enforcement: (optional), Specifies the enforcement level of
+            the status checks. Must be one of 'off', 'non_admins', or
+            'everyone'. Set to None to leave unchanged.
+        :param list status_checks: (optional), An iterable of strings naming
+            status checks that must pass before merging. Set to None to leave
+            unchanged.
+        """
+        enable = self.protection['enabled'] if enable is None else bool(enable)
```

Comment:
Github expects via json `true` or `false`. 500 errors appear to occur if anything else is sent, thus if a github3.py user happens to put any non-boolean value (e.g., `1`, 'on`), this will provide some resiliency. If it's more consistent for the caller to get it right, then I'm happy to remove the casting.